lib/XMLHttpRequest.js

```diff
@@ -365,8 +410,21 @@ function XMLHttpRequest(opts) {
       throw new Error("INVALID_STATE_ERR: send has already been called");
     }
 
-    var ssl = false, local = false;
-    var url = Url.parse(settings.url);
+    var ssl = false, local = false, datauri = false;
```

Comment:
If we name these like the flags they are, determining their value type elsewhere will be much easier: `isSsl`, `isLocal`, `isDataUri`.

Reply:
the `ssl` and `local` are from old implementation, i just simply add `datauri` to match with them. However if you want i can refactor these.